src/main/java/org/sonarlint/intellij/inspection/SonarLintGlobalInspection.java

```diff
@@ -141,20 +144,68 @@ public void visitFile(PsiFile psiFile) {
   }
 
   ProblemDescriptor getProblemDescriptor(Issue issue, PsiFile psiFile, @NotNull InspectionManager manager, boolean isLocal) {
-    if (issue.getLine() != null) {
-      return manager.createProblemDescriptor(psiFile,
-        InspectionUtils.getLineRange(psiFile, issue.getLine()),
+    Integer startLine = issue.getStartLine();
+    if (startLine == null) {
+      // No line, global issue
+      return createProblemDescriptorFileLevel(issue, psiFile, manager, isLocal);
+    }
+    Integer startLineOffset = issue.getStartLineOffset();
+    if (startLineOffset == null) {
+      // No offset, either old SQ version or language plugin doesn't provide data. Fallback to text range
+      return createProblemDescriptorPreciseTextRange(issue, psiFile, manager, isLocal, InspectionUtils.getLineRange(psiFile, startLine));
+    }
+    Integer endLine = issue.getEndLine();
+    Integer endLineOffset = issue.getEndLineOffset();
+    int ijStartLine = startLine - 1;
+    int ijEndLine = endLine - 1;
+    final Document document = PsiDocumentManager.getInstance(psiFile.getProject()).getDocument(psiFile);
+    PsiElement startElement;
+    PsiElement endElement;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused "endElement" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1481)